klippy/extras/temperature_probe.py

```diff
@@ -0,0 +1,375 @@
+            toolhead.manual_move(self.cal_pos, move_speed)
+        elif self.cal_extruder_temp is not None:
+            cur_pos[2] = self.cal_extruder_z
+            toolhead.manual_move(cur_pos, move_speed)
+            self._set_extruder_temp(self.cal_extruder_temp, True)
+
+    def _get_speeds(self):
+        pparams = self._get_probe().get_probe_params()
+        probe_speed = pparams["probe_speed"]
+        lift_speed = pparams["lift_speed"]
+        move_speed = self.speed or max(probe_speed, lift_speed)
+        return lift_speed, probe_speed, move_speed
+
+    cmd_PROBE_DRIFT_CALIBRATE_help = (
+        "Calibrate probe temperature drift compensation"
+    )
+    def cmd_PROBE_DRIFT_CALIBRATE(self, gcmd):
+        if self.cal_helper is None:
+            raise gcmd.error(
+                "No calibration helper registered for [%s]"
+                % (self.name,)
+            )
+        self._check_homed()
+        probe = self._get_probe()
+        probe_name = probe.get_status(None)["name"]
+        short_name = probe_name.split(maxsplit=1)[-1]
+        if short_name != self.name.split(maxsplit=1)[-1]:
+            raise self.gcode.error(
+                "[%s] not linked to registered probe [%s]."
+                % (self.name, probe_name)
+            )
+        manual_probe.verify_no_manual_probe(self.printer)
+        if self.in_calibration:
+            raise gcmd.error(
+                "Already in probe drift calibration. Use "
+                "PROBE_DRIFT_COMPLETE or ABORT to exit."
+            )
+        cur_temp = self.smoothed_temp
+        target_temp = gcmd.get_float("TARGET", above=cur_temp)
+        step = gcmd.get_float("STEP", 2., minval=1.0)
+        expected_count = int(
+            (target_temp - cur_temp) / step + .5
+        )
+        if expected_count < 3:
+            raise gcmd.error(
+                "Invalid STEP and/or TARGET parameters resulted "
+                "in too few expected samples: %d"
+                % (expected_count,)
+            )
+        try:
+            self.gcode.register_command(
+                "PROBE_DRIFT_NEXT", self.cmd_PROBE_DRIFT_NEXT,
+                desc=self.cmd_PROBE_DRIFT_NEXT_help
+            )
+            self.gcode.register_command(
+                "PROBE_DRIFT_COMPLETE", self.cmd_PROBE_DRIFT_COMPLETE,
+                desc=self.cmd_PROBE_DRIFT_NEXT_help
+            )
+        except self.printer.config_error:
+            raise gcmd.error(
+                "Auxiliary Probe Drift Commands already registered. Use "
+                "PROBE_DRIFT_COMPLETE or ABORT to exit."
+            )
+        self.in_calibration = True
+        self.cal_helper.start_calibration()
+        self.target_temp = target_temp
+        self.step = step
+        self.sample_count = 0
+        self.expected_count = expected_count
+        # If configured move to heating position and turn on extruder
+        try:
+            self._move_to_start()
+        except Exception:
```

Comment:
Very minor, but this should probably be `except self.printer.config_error` (as above).  An internal coding error probably shouldn't be caught here.